lib/project-data.ts

```diff
@@ -39,11 +41,62 @@ export class ProjectData implements IProjectData {
 								suppressCommandHelp: true},
 							this.projectFilePath, err.toString());
 					}
-				}
+
+					if(data) {
+						this.projectId = data.id;
+					} else { // This is the case when we have package.json file but nativescipt key is not presented in it
+						this.tryToUpgradeProject().wait();
+					}
+				} 
+			} else { // This is the case when no project file found
+				this.tryToUpgradeProject().wait();
+			}
+		}).future<void>()();
+	}
+
+	private throwNoProjectFoundError(): void {
+		this.$errors.fail("No project found at or above '%s' and neither was a --path specified.", process.cwd());
+	}
+
+	private tryToUpgradeProject(): IFuture<void> {
+		return (() => {
+			var projectDir = this.projectDir || path.resolve(options.path || ".");
+			var oldProjectFilePath = path.join(projectDir, ProjectData.OLD_PROJECT_FILE_NAME);
+			if(this.$fs.exists(oldProjectFilePath).wait()) {
+				this.upgrade(projectDir, oldProjectFilePath).wait();
 			} else {
-				this.$errors.fail("No project found at or above '%s' and neither was a --path specified.", process.cwd());
+				this.throwNoProjectFoundError();
 			}
 		}).future<void>()();
 	}
+
+	private upgrade(projectDir: string, oldProjectFilePath: string): IFuture<void> {
+		return (() => {
+			try {
+				var oldProjectData = this.$fs.readJson(oldProjectFilePath).wait();
+
+				var newProjectFilePath = this.projectFilePath || path.join(projectDir, this.$staticConfig.PROJECT_FILE_NAME);
+				var newProjectData = this.$fs.exists(newProjectFilePath).wait() ? this.$fs.readJson(newProjectFilePath).wait() : {};
+				newProjectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE] = oldProjectData;
+				this.$fs.writeJson(newProjectFilePath, newProjectData).wait();
+
+				this.$fs.deleteFile(oldProjectFilePath).wait();
+			} catch(err) {
+				this.$logger.out("An error occurred while upgrading your project.");
```

Comment:
Consider showing the err.Message to the user to help them troubleshoot the issue.